nalgebra-sparse/src/cs.rs

```diff
@@ -543,3 +544,146 @@ pub fn convert_counts_to_offsets(counts: &mut [usize]) {
         offset += count;
     }
 }
+
+/// Validates cs data, sorts minor indices and values
+pub(crate) fn validate_and_optionally_sort_cs_data<T>(
+    major_dim: usize,
+    minor_dim: usize,
+    major_offsets: &[usize],
+    minor_indices: &mut [usize],
+    values: Option<&mut [T]>,
+    sort: bool,
+) -> Result<(), SparseFormatError>
+where
+    T: Scalar,
+{
+    let mut value_refs: &mut [T] = &mut Vec::new();
+    match values {
+        Some(values) => {
+            if minor_indices.len() != values.len() {
+                return Err(SparseFormatError::from_kind_and_msg(
+                    SparseFormatErrorKind::InvalidStructure,
+                    "Number of values and minor indices must be the same",
+                ));
+            }
+            value_refs = values;
+        }
+        None => {
+            if sort && minor_indices.len() > 0 {
+                return Err(SparseFormatError::from_kind_and_msg(
+                    SparseFormatErrorKind::InvalidStructure,
+                    "No values provided for sorting.",
```

Comment:
It seems to me that it would make total sense to sort without `values` if we're constructing e.g. a `SparsityPattern` from unsorted indices. But I guess this is not available yet, right? So it's fine that we have this restriction for now!

I think it's also OK to panic in this case instead of returning an error, because this is an *internal error* - we're the only one who can call this method, and it's an error on our part if this path gets triggered. (So could just replace with an assert)